ibm/resource_ibm_cis_origin_pool.go

```diff
@@ -0,0 +1,252 @@
+				Computed: true,
+			},
+		},
+
+		Create: resourceCISpoolCreate,
+		Read:   resourceCISpoolRead,
+		Update: resourceCISpoolUpdate,
+		Delete: resourceCISpoolDelete,
+	}
+}
+
+func resourceCISpoolCreate(d *schema.ResourceData, meta interface{}) error {
+	cisClient, err := meta.(ClientSession).CisAPI()
+	log.Printf("   client %v\n", cisClient)
+	if err != nil {
+		return err
+	}
+
+	cisId := d.Get("cis_id").(string)
+	name := d.Get("name").(string)
+	origins := d.Get("origins").(*schema.Set)
+	checkRegions := d.Get("check_regions").(*schema.Set).List()
+
+	var pools *[]v1.Pool
+	pools, err = cisClient.Pools().ListPools(cisId)
+	if err != nil {
+		log.Printf("ListPools Failed %s\n", err)
+		return err
+	}
+
+	var poolNames []string
+	poolsObj := *pools
+	for _, pool := range poolsObj {
+		poolNames = append(poolNames, pool.Name)
+	}
+	var pool *v1.Pool
+	var poolObj v1.Pool
+
+	index := indexOf(name, poolNames)
+	if index == -1 {
+
+		poolNew := v1.PoolBody{}
+		poolNew.Name = name
+		poolNew.CheckRegions = expandToStringList(checkRegions)
+		poolNew.Origins = expandOrigins(origins)
+
+		if notEmail, ok := d.GetOk("notification_email"); ok {
+			poolNew.NotEmail = notEmail.(string)
+		}
+		if monitor, ok := d.GetOk("monitor"); ok {
+			poolNew.Monitor = monitor.(string)
+		}
+		if enabled, ok := d.GetOk("enabled"); ok {
+			poolNew.Enabled = enabled.(bool)
+		}
+		if minOrigins, ok := d.GetOk("minimum_origins"); ok {
+			log.Printf("Was min origins set?: >>>>>%v<<<<<\n", minOrigins)
+			poolNew.MinOrigins = minOrigins.(int)
+		}
+		if description, ok := d.GetOk("description"); ok {
+			poolNew.Description = description.(string)
+		}
+
+		pool, err = cisClient.Pools().CreatePool(cisId, poolNew)
+		if err != nil {
+			log.Printf("CreatePools Failed %s\n", err)
+			return err
+		}
+		poolObj = *pool
+	} else {
+		return fmt.Errorf("Resource with name %s already exists", name)
+	}
+
+	d.SetId(poolObj.Id)
+	d.Set("name", poolObj.Name)
+
+	return resourceCISpoolRead(d, meta)
+}
+
+func resourceCISpoolRead(d *schema.ResourceData, meta interface{}) error {
+	cisClient, err := meta.(ClientSession).CisAPI()
+	if err != nil {
+		return err
+	}
+
+	var poolId string
+
+	poolId = d.Id()
+	cisId := d.Get("cis_id").(string)
+
+	log.Printf("resourceCISpoolRead - Getting Pool %v\n", poolId)
+	var pool *v1.Pool
+
+	pool, err = cisClient.Pools().GetPool(cisId, poolId)
+	if err != nil {
+		log.Printf("resourceCIpoolRead - ListPools Failed %s\n", err)
+		return err
+	} else {
+		log.Printf("resourceCISpoolRead - Retrieved Pool %v\n", pool)
+
+		poolObj := *pool
+		d.Set("name", poolObj.Name)
+		d.Set("check_regions", poolObj.CheckRegions)
+		d.Set("origins", poolObj.Origins)
+		d.Set("description", poolObj.Description)
+		d.Set("enabled", poolObj.Enabled)
+		d.Set("minimum_origins", poolObj.MinOrigins)
+		d.Set("monitor", poolObj.Monitor)
+		d.Set("notification_email", poolObj.NotEmail)
+
+	}
+	return nil
+}
+
+func expandToStringList(list interface{}) []string {
```

Comment:
We already have func expandStringList in structures.go

Reply:
Changed to expandStringList